resources/app.rb

```diff
@@ -4,39 +4,52 @@
 default_action :create
 
 property :user, String, default: lazy { name }
-property :service_after, Array, default: %w(network.target)
+property :service_after, String, default: 'network.target'
 property :service_name, String, default: lazy { name }
 property :service_type, String, default: 'forking'
+property :service_wants, [String, nil]
 property :start_cmd, String, required: true
 property :reload_cmd, String, default: '/bin/kill -USR2 $MAINPID'
 property :description, [String, nil]
-property :environment, Hash, default: {}
-property :pid_file, String, required: true
+property :environment, [String, nil]
+property :pid_file, [String, nil]
 property :wanted_by, String, default: 'multi-user.target'
 property :working_directory, [String, nil]
 property :environment_file, [String, nil]
+property :verify, [true, false], default: false
 
 action :create do
   sudo new_resource.user do
     user new_resource.user
     commands osl_sudo_commands(new_resource.service_name)
     nopasswd true
+    action :nothing
   end
 
-  systemd_service new_resource.service_name do
-    unit_description new_resource.description unless new_resource.description.nil?
-    unit_after new_resource.service_after
-    install_wanted_by new_resource.wanted_by
-    service_type new_resource.service_type
-    service_user new_resource.user
-    service_environment new_resource.environment
-    service_environment_file new_resource.environment_file unless new_resource.environment_file.nil?
-    service_working_directory new_resource.working_directory unless new_resource.working_directory.nil?
-    service_pid_file new_resource.pid_file
-    service_exec_start new_resource.start_cmd
-    service_exec_reload new_resource.reload_cmd
-    verify false
+  systemd_unit "#{new_resource.service_name}.service" do
+    content({
+      'Unit' => {
+        'Description' => new_resource.description,
+        'After' => new_resource.service_after,
+        'Wants' => new_resource.service_wants,
```

Comment:
If `service_wants` is nil, we shouldn't be including this at all. Right now it's setting `Wants = ` in the file which is incorrect I think. The same thing happens with other parameters.

Reply:
The simplest way to do this would be to remove any nil entries instead of guarding everything:
```rb
irb> h
{"Unit"=>{"Description"=>"desc", "After"=>nil, "Wants"=>nil},
 "Service"=>{"Type"=>nil, "User"=>"user"},
 "Install"=>{"WantedBy"=>"warrant"}}

irb> h.transform_values(&:compact)
{"Unit"=>{"Description"=>"desc"}, "Service"=>{"User"=>"user"}, "Install"=>{"WantedBy"=>"warrant"}}
```